tests/utils/common/common.py

```diff
@@ -161,46 +220,34 @@ def run_after_connect(cmd, conn, wait=360):
     orig_timeout = conn.connect_timeout
     conn.connect_timeout = 60
     timeout = time.time() + 60 * 3
+    latest_exc = None
 
-    while time.time() < timeout:
-        try:
-            print("will try to connect to host", conn.host)
-            result = conn.run(cmd, hide=True)
-            return result.stdout
-        except NoValidConnectionsError as e:
-            print("Could not connect to host %s: %s" % (conn.host, e))
-            time.sleep(30)
-            continue
-        except SSHException as e:
-            print("Got SSH exception while connecting to host %s: %s" % (conn.host, e))
-            if not (
-                "Connection reset by peer" in str(e)
-                or "Error reading SSH protocol banner" in str(e)
-                or "No existing session" in str(e)
-            ):
-                raise e
-            time.sleep(30)
-            continue
-        except OSError as e:
-            # The OSError is happening while there is no QEMU instance initialized
-            print(
-                "Got OSError exception while connecting to host %s: %s" % (conn.host, e)
-            )
-            if not "Cannot assign requested address" in str(e):
+    try:
+        while time.time() < timeout:
+            try:
+                print("will try to connect to host", conn.host)
+                result = conn.run(cmd, hide=True)
+                return result.stdout
+            except ConnectionError as e:
+                latest_exc = e
+                print(
+                    "Got SSH exception while connecting to host %s: %s" % (conn.host, e)
+                )
+                time.sleep(30)
+                continue
+            except Exception as e:
```

Comment:
S: Maybe move the catchall out of the loop, into the outermost `try` statement?

Reply:
I think I prefer it in the inner one, mostly for readability reasons. The outer one is not doing any error checking, it exists only to set the timeout back to what it was (I was looking for ways to avoid it altogether actually, but couldn't find any).